source/code/providers/Container_ContainerLog_Class_Provider.cpp

```diff
@@ -109,7 +109,8 @@ class ContainerLogQuery
         deque<Container_ContainerLog_Class> result;
 
         string logDriver = getLogDriverName();
-        if (logDriver.compare("json-file") != 0)
+        //Docker rest api endpoint supports both json-file and journald logging drivers
+        if ((logDriver.compare("json-file") != 0) && (logDriver.compare("journald") != 0))
```

Comment:
If its neither then it returns immediately without doing any log collection since we can only support json-file or journald, else it continues with the log parsing